apis/stacks/v1alpha1/types.go

```diff
@@ -118,6 +133,16 @@ func (si *ClusterStackInstall) SetConditions(c ...runtimev1alpha1.Condition) {
 	si.Status.SetConditions(c...)
 }
 
+// GetSpec gets the Spec of the StackInstall
+func (si *StackInstall) GetSpec() *StackInstallSpec {
```

Comment:
`GetSpec` is borrowed from #1264 and will be removed from this PR if that is merged first.

Reply:
I've borrowed a few more features from #1264 for modifying the unstructured Stack and StackDefinition resources. 

These will need to be rebased here once that is merged. 

In this PR, I've applied an improved pattern for making these modifications:
https://github.com/crossplane/crossplane/pull/1281/files#diff-46f16c3aed495fb04c3696c537ce7501R233-R437